### PR TITLE
fix: Support for OUs without accounts

### PR DESCRIPTION
**Pull request by moss2k13**

### CODE REVIEW

File: modules/organizational-unit/main.tf [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
- * This module creates AWS Organization OUs and uses AWS Accounts module to 
+ * This module creates AWS Organization OUs and uses AWS Accounts module to
  * create accounts inside of that OU.
  *
  * It also provides functionality to attach a list of organizational policies to the OU.
  */
 
 
-resource "aws_organizations_organizational_unit" "this_orgranizations_organizational_unit" {
+resource "aws_organizations_organizational_unit" "this_organizations_organizational_unit" {
```
Comment (dgniewek) [inline modules/organizational-unit/main.tf:11]: Unfortunately changing that names (spelling error) is valid but causes a breaking change and will introduce problems with existing terraform state

File: modules/organizational-unit/main.tf [hunk 2]
```diff
@@ -20,20 +20,20 @@ resource "aws_organizations_policy_attachment" "this_organizations_policy_attach
   count = length(var.attached_policies)
 
   policy_id = var.attached_policies[count.index]
-  target_id = aws_organizations_organizational_unit.this_orgranizations_organizational_unit.id
+  target_id = aws_organizations_organizational_unit.this_organizations_organizational_unit.id
```
Comment (dgniewek) [inline modules/organizational-unit/main.tf:23]: The same comment

File: modules/organizational-unit/main.tf [hunk 3]
```diff
@@ -20,20 +20,20 @@ resource "aws_organizations_policy_attachment" "this_organizations_policy_attach
...
 }
 
 # if list of accounts is added to OU - create them inside that OU
-module "this_orgranizations_organizational_unit_account" {
-  for_each = var.accounts
+module "this_organizations_organizational_unit_account" {
```
Comment (dgniewek) [inline modules/organizational-unit/main.tf:27]: The same comment

File: modules/organizational-unit/locals.tf [hunk 4]
```diff
@@ -0,0 +1,3 @@
+locals {
+  accounts = coalesce(var.accounts, {})
```
Comment (dgniewek) [inline modules/organizational-unit/locals.tf:2]: This seems fine - and solves the problem, but maybe migration to newer terraform and setting proper object format would be a better way to go?

**Comment by moss2k13** [comment]
Closing without merging: mentioned changes will be added to a separate pull request related with migrating to terraform version >= 1.3

**Pull request closed without merging**